java/core/src/test/java/com/microsoft/bond/BondedTest.java

```diff
@@ -70,4 +73,31 @@ public void bondedEquality() throws IOException {
             }
         }
     }
+
+    @Test
+    public void covariance() throws IOException {
```

Comment:
Consider adding a test case for the throwing behavior as well. E.g., a downcast and an unrelated type.